transport/transport.go

```diff
@@ -168,6 +168,9 @@ type Stream struct {
 	id uint32
 	// nil for client side Stream.
 	st ServerTransport
+	// Keep the user context for stats handling.
+	// All stats handling should use the user context instead of the stream context.
+	userCtx context.Context
```

Comment:
I would suggest renaming it to "statsCtx" because "userCtx" is not appropriate name on server side.